### PR TITLE
WIP - Add --help and --list-mode to Plugin Packs documentation 🚀 

### PR DESCRIPTION
**Pull request by itoussies**
## Description

⚠️ Please disregard everything except the FAQ section ⚠️

More testing need to be done but here is a early first draft. We now have the plugin options (global options to all modes are missing for the moment) and all plugin modes (even those not in the Plugin Pack) in the new documentation template

## Target version

- [ ] 21.10.x (staging)
- [ ] 22.04.x (staging)
- [ ] 22.10.x (staging)
- [ ] Cloud (staging)
- [X] Plugin Packs (staging)
- [ ] 23.04.x (next)

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 1]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-dir        | Directory for statefile (Default: '/var/lib/centreon/centplugins').                                                                                  |
+| --statefile-suffix     | Add a suffix for the statefile name (Default: '').                                                                                                   |
+| --statefile-concat-cwd | Concat current working directory with option '--statefile-dir'. Useful on Windows when plugin is compiled.                                           |
+| --statefile-format     | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                   |
+| --statefile-key        | Key to encrypt/decrypt cache.                                                                                                                        |
+| --statefile-cipher     | Cipher to encrypt cache (Default: 'AES').                                                                                                            |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]: These options are not related to cpu-detailed, they are available for all plugins/modes.

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 2]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --memexpiration         | Time to keep data in seconds (Default: 86400).                                                                                           |
+| --statefile-dir         | Directory for statefile (Default: '/var/lib/centreon/centplugins').                                                                      |
+| --statefile-suffix      | Add a suffix for the statefile name (Default: '').                                                                                       |
+| --statefile-concat-cwd  | Concat current working directory with option '--statefile-dir'. Useful on Windows when plugin is compiled.                               |
+| --statefile-format      | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                       |
+| --statefile-key         | Key to encrypt/decrypt cache.                                                                                                            |
+| --statefile-cipher      | Cipher to encrypt cache (Default: 'AES').                                                                                                |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:418]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md (hunk 1 above)
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 3]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-suffix     | Add a suffix for the statefile name (Default: '').                                                                                                                                     |
+| --statefile-concat-cwd | Concat current working directory with option '--statefile-dir'. Useful on Windows when plugin is compiled.                                                                             |
+| --statefile-format     | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                                                     |
+| --statefile-key        | Key to encrypt/decrypt cache.                                                                                                                                                          |
+| --statefile-cipher     | Cipher to encrypt cache (Default: 'AES').                                                                                                                                              |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:457]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 4]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-format       | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                                                                                                                                                         |
+| --statefile-key          | Key to encrypt/decrypt cache.                                                                                                                                                                                                                                                              |
+| --statefile-cipher       | Cipher to encrypt cache (Default: 'AES').                                                                                                                                                                                                                                                  |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:529]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 5]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-suffix     | Add a suffix for the statefile name (Default: '').                                                                                                                                       |
+| --statefile-concat-cwd | Concat current working directory with option '--statefile-dir'. Useful on Windows when plugin is compiled.                                                                               |
+| --statefile-format     | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                                                       |
+| --statefile-key        | Key to encrypt/decrypt cache.                                                                                                                                                            |
+| --statefile-cipher     | Cipher to encrypt cache (Default: 'AES').                                                                                                                                                |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:578]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 6]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-format       | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                                                                                                                                                         |
+| --statefile-key          | Key to encrypt/decrypt cache.                                                                                                                                                                                                                                                              |
+| --statefile-cipher       | Cipher to encrypt cache (Default: 'AES').                                                                                                                                                                                                                                                  |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:636]:
```suggestion
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 7]
```diff
@@ -6,240 +6,695 @@ import Tabs from '@theme/Tabs';
...
+| --statefile-dir        | Directory for statefile (Default: '/var/lib/centreon/centplugins').                                                                                       |
+| --statefile-suffix     | Add a suffix for the statefile name (Default: '').                                                                                                        |
+| --statefile-concat-cwd | Concat current working directory with option '--statefile-dir'. Useful on Windows when plugin is compiled.                                                |
+| --statefile-format     | Format used to store cache (can be: 'dumper', 'storable', 'json').                                                                                        |
+| --statefile-key        | Key to encrypt/decrypt cache.                                                                                                                             |
+| --statefile-cipher     | Cipher to encrypt cache (Default: 'AES').                                                                                                                 |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:685]:
```suggestion
```

**Review by omercier — CHANGES_REQUESTED** [review]
Some options should be removed. I entered commit suggestions for the french version only but they apply to the english version as well.
This is an awesome step forward for plugins documentation :star_struck:

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md (hunk 1 above)
Comment (itoussies) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]: Those options are not available for the swap mode for example. The ones you don't see yet are available for all modes.  
(I need to write a global tab for that). Maybe I can include another tab that says: "Thoses options are available for the following modes:...."

Comment (omercier) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]: Ok, my bad. It's fine this way.

Comment (omercier) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]: Or maybe is it possible to indicate which options are technical (the cache/statefile etc) which are generic (change-perfdata etc. which is not present in the generated doc) and which are mode-specific ?

Comment (itoussies) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:398]: What do you think about something like this ?

new_doc

**Comment by itoussies** [comment]
Closed because the template is in prod now. Next enhancements/changes will be shown internally soon.
